lib/data/hive/hive_collection_network_database.dart

```diff
@@ -0,0 +1,139 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:fluffychat/data/hive/hive_constants.dart';
+import 'package:fluffychat/data/hive/network/network_connectivity_hive_obj.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/flutter_hive_collections_database.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:hive/hive.dart';
+import 'package:matrix/matrix.dart';
+import 'package:path_provider/path_provider.dart';
+
+class HiveCollectionNetworkDatabase {
+  final String name;
+  final String? path;
+  final HiveCipher? key;
+
+  late BoxCollection _collection;
+
+  String get _networkConnectivityBoxName => 'network_connectivity_box';
+  late CollectionBox<NetworkConnectivityHiveObj> networkConnectivityBox;
+
+  HiveCollectionNetworkDatabase(this.name, this.path, {this.key});
+
+  static Future<HiveCollectionNetworkDatabase> databaseBuilder() async {
+    Logs().d('Open Hive for Network Connectivity...');
+    HiveAesCipher? hiverCipher;
+    try {
+      // Workaround for secure storage is calling Platform.operatingSystem on web
+      if (kIsWeb) throw MissingPluginException();
+
+      const secureStorage = FlutterSecureStorage();
+      final containsEncryptionKey =
+          await secureStorage.read(key: FlutterHiveCollectionsDatabase.cipherStorageKey) != null;
+      if (!containsEncryptionKey) {
+        // do not try to create a buggy secure storage for new Linux users
+        if (Platform.isLinux) throw MissingPluginException();
+        final key = Hive.generateSecureKey();
+        await secureStorage.write(
+          key: FlutterHiveCollectionsDatabase.cipherStorageKey,
+          value: base64UrlEncode(key),
+        );
+      }
+
+      // workaround for if we just wrote to the key and it still doesn't exist
+      final rawEncryptionKey = await secureStorage.read(key: FlutterHiveCollectionsDatabase.cipherStorageKey);
+      if (rawEncryptionKey == null) throw MissingPluginException();
+
+      hiverCipher = HiveAesCipher(base64Url.decode(rawEncryptionKey));
+    } on MissingPluginException catch (_) {
+      const FlutterSecureStorage()
+          .delete(key: FlutterHiveCollectionsDatabase.cipherStorageKey)
+          .catchError((_) {});
+      Logs().i('Hive encryption is not supported on this platform');
+    } catch (e, s) {
+      const FlutterSecureStorage()
+          .delete(key: FlutterHiveCollectionsDatabase.cipherStorageKey)
+          .catchError((_) {});
```

Comment:
duplicate lines 52-54. 

- We can write func handle this case and can be reuse it? 
- What do you think?